hicexplorer/hicAggregateContacts.py

```diff
@@ -187,14 +227,187 @@ def read_bed_per_chrom(fh):
+
+    bin_id1 = ma.getRegionBinRange(toString(chrom1), start1, end1)
+    bin_id2 = ma.getRegionBinRange(toString(chrom2), start2, end2)
+    if bin_id1 == bin_id2:
+        return
+    if (bin_id1 is None) or (bin_id2 is None):
+        return
+    else:  # If the regions size is bigger than a bin then:
+        if largeRegionsOperation == 'first':
+            bin_id1 = bin_id1[0]
+            bin_id2 = bin_id2[0]
+        elif largeRegionsOperation == 'last':
+            bin_id1 = bin_id1[-1]
+            bin_id2 = bin_id2[-1]
+        elif largeRegionsOperation == 'center':
+            bin_id1 = int(np.floor(np.mean(bin_id1)))
+            bin_id2 = int(np.floor(np.mean(bin_id2)))
+    if bin_id1 > bin_id2:
+        if chrom1 == chrom2:
+            bin_id1, bin_id2 = sorted([bin_id1, bin_id2])
+        else:
+            chr1, str1, en1, bin1 = chrom1, start1, end1, bin_id1
+            chrom1, start1, end1, bin_id1 = chrom2, start2, end2, bin_id2
+            chrom2, start2, end2, bin_id2 = chr1, str1, en1, bin1
+    agg_info["counter"] += 1
+    if agg_info["counter"] % 50000 == 0:
+        log.info("Number of contacts considered: {:,}".format(agg_info["counter"]))
+
+    bin_size = ma.getBinSize()
+    chrom1_bin_range = ma.getChrBinRange(toString(chrom1))
+    chrom2_bin_range = ma.getChrBinRange(toString(chrom2))
+
+    if chrom1 == chrom2:  # chrom1 == chrom2 can happen in intra or all
+        if (chrom1 not in agg_info["agg_total"]) and (mode == "intra-chr") and (perChr == True):
+            agg_info["agg_total"][chrom1] = 0
+            agg_info["agg_matrix"][chrom1] = []
+            agg_info["agg_diagonals"][chrom1] = []
+            agg_info["agg_contact_position"][chrom1] = []
+            agg_info["agg_center_values"][chrom1] = []
+
+        min_dist, max_dist = range.split(":")
+        min_dist_in_bins = int(min_dist) // bin_size
+        max_dist_in_bins = int(max_dist) // bin_size
+        if (min_dist_in_bins > abs(bin_id2 - bin_id1)) or (abs(bin_id2 - bin_id1) > max_dist_in_bins):
+            return
+    if (bin_id1, bin_id2) in agg_info["seen"]:
+        return
+    agg_info["seen"].append((bin_id1, bin_id2))
+    if bin_id1 - M_half < chrom1_bin_range[0] or bin_id1 + M_half >= chrom1_bin_range[1]:
+        log.info("The given interval exceeds the chromosome range on {}. It is skipped.".format(chrom1))
+        return
+
+    if bin_id2 - M_half < chrom2_bin_range[0] or bin_id2 + M_half >= chrom2_bin_range[1]:
+        log.info("The given interval exceeds the chromosome range on {}. It is skipped.".format(chrom2))
+        return
+    try:
+        mat_to_append = ma.matrix[bin_id1 - M_half:bin_id1 + M_half + 1, :][:, bin_id2 - M_half:bin_id2 + M_half + 1].todense().astype(float)
+    except IndexError:
+        log.info("index error for {} {}".format(bin_id1, bin_id2))
+        return
+
+    if mat_to_append.sum() == 0:
+        agg_info["empty_mat"] += 1
+        return
+
+    agg_info["counter"] += 1
```

Comment:
line 338 you already incremented the counter... They should be different counters or incremented only once.